tools/_setup_generation/step_designer.py

```diff
@@ -0,0 +1,73 @@
+# ##########################################################################################
+# Step to generate the documentation pages for Taipy Designer, after they were copied
+# locally from the taipy-designer repository.
+#
+# ##########################################################################################
+
+import os
+import re
+from io import StringIO
+
+from .setup import SetupStep, Setup
+
+
+class DesignerStep(SetupStep):
+    PREFIX = "userman/ecosystem/designer"
+    def __init__(self):
+        self.navigation = ""
+
+    def get_id(self) -> str:
+        return "designer"
+
+    def get_description(self) -> str:
+        return "Retrieve the Designer documentation files."
+
+    def setup(self, setup: Setup): ...
+
+    def enter(self, setup: Setup):
+        self.DESIGNER_PATH = os.path.join(
+            setup.docs_dir, *DesignerStep.PREFIX.split("/")
+        )
+        self.MKDOCS_TMPL = os.path.join(self.DESIGNER_PATH, "mkdocs.yml_template")
+        if not os.access(self.MKDOCS_TMPL, os.R_OK):
+            raise FileNotFoundError(
+                f"FATAL - Could not read docs/{DesignerStep.PREFIX}/mkdocs.yml_template"
+            )
+        self._read_mkdocs_template()
```

Comment:
Why not return the navigation and assign it here to increase readability?

```suggestion
        self.navigation = self._read_mkdocs_template()
```
That's a way to make the output of this method explicit. It is easier to understand this method computes the navigation without having to read the code.

What do you think?

Reply:
Agreed!